### PR TITLE
[SDK/CLI] Add basic orchestrator test

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_submitter/experiment_orchestrator.py [hunk 1]
```diff
@@ -116,7 +116,8 @@ def _run_flow_node(self, node, experiment, run_dict):
             variant=node.variant,
             flow=node.path,
             connections=node.connections,
-            experiment_name=experiment.name,
+            # Experiment name can't work for now, it is used only for show cloud exp name
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_submitter/experiment_orchestrator.py:119]: maybe just remove this? as if we want to add back by using experiment.name? experiment is already an parameter to ExeprimentRun

File: src/promptflow/tests/sdk_cli_test/conftest.py [hunk 2]
```diff
@@ -83,6 +87,21 @@ def setup_local_connection(local_client, azure_open_ai_connection):
     _connection_setup = True
 
 
+@pytest.fixture
+def setup_experiment_table(local_client, azure_open_ai_connection):
```
Comment (zhengfeiwang) [inline src/promptflow/tests/sdk_cli_test/conftest.py:91]: why we need `local_client` here?

Comment (brynn-code) [reply src/promptflow/tests/sdk_cli_test/conftest.py:91]: Oh, I was wrong about that, it's not required here.

**Review by D-W- — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_test/e2etests/test_experiment.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        assert exp_get._to_dict() == exp._to_dict()
+
+    @pytest.mark.usefixtures("use_secrets_config_file", "recording_injection", "setup_local_connection")
+    def test_experiment_start(self):
+        template_path = EXP_ROOT / "basic-no-script-template" / "basic.exp.yaml"
+        # Load template and create experiment
+        template = load_common(ExperimentTemplate, source=template_path)
+        experiment = Experiment.from_template(template)
+        client = PFClient()
+        exp = client._experiments.create_or_update(experiment)
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_experiment.py:72]: Is there a way to start an expriment from experiment template directly? like `client._experiment.create_or_update(experiment_template="xxx")`

File: src/promptflow/promptflow/_sdk/schemas/_experiment.py [hunk 4]
```diff
@@ -30,6 +36,11 @@ class Meta:
     inputs = fields.Dict(keys=fields.Str)
     path = UnionField([LocalPathField(required=True), fields.Str(required=True)])
 
+    @pre_load
+    def warning_unknown_fields(self, data, **kwargs):
```
Comment (0mza987) [inline src/promptflow/promptflow/_sdk/schemas/_experiment.py:40]: What unknown fileds for node that does not need warning? Besides, this implementation is kinda werid, we should set `unknown="exclude"` at the place where the schema being used, or you change the original `warning_unknown_fields` function to skip for non-runschema instances?

**Pull request merged**